src/makeCalculator.js

```diff
@@ -4,7 +4,30 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add: (num) => (calculator.result += num),
+    subtract: (num) => (calculator.result -= num),
+    multiply: (num) => (calculator.result *= num),
+    divide: (num) => {
+      if (num === 0) {
+        return;
+      }
+      calculator.result /= num;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(operator, num) {
+      operator(num);
```

Comment:
The `operate` method assumes that the `operator` parameter is a function. It would be safer to check if `operator` is indeed a function before calling it, to prevent runtime errors.